git-submodules-status

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+
+################################################################################
+#
+#   Description:
+#   ------------
+#       Fetches all submodules from their respective remote repositories, and
+#       shows a table that lists for each submodule : the latest available
+#       version, the current version and a small visual cue to easily identify
+#       mismatches between the two.
+#
+#   Note that this script assumes that a project's submodules are present in a
+#   directory called "submodules" in the project's top-level git directory.
+#
+#   Usage:
+#   ------
+#   (first place this script in one of the directories in $PATH)
+#       $> cd path/to/repo
+#       $> git submodules-status
+#
+################################################################################
+
+appendSpaces()
+{
+    INITIAL_STR=$1
+    TARGET_LENGTH=$2
+
+    SPACES_TO_APPEND=$((TARGET_LENGTH - ${#INITIAL_STR}))
+
+    echo -n $INITIAL_STR
+    for ((i = 0; i < $SPACES_TO_APPEND; i++))
+    do
+        echo -n " "
+    done
+    echo ""
+}
+
+# Confirm that we are in a git repository
+GIT_TOP=$(git rev-parse --show-toplevel) || exit $?
+
+# Check if the project has any submodules in the first place
+if ! [ -a $GIT_TOP/.gitmodules ]; then
+    PROJECT_NAME=$(basename $GIT_TOP)
+    echo "Project '$PROJECT_NAME' has no submodules. Nothing to do here!"
+    exit 0
+fi
+
+# Confirm that all the submodules have been initialised
+UNINITIALISED_SUBMODS=(`git submodule | grep -v "^ " | awk -F"/" '{print $NF}'`)
+if [ ${#UNINITIALISED_SUBMODS[@]} -gt 0 ]; then
+    echo "The following submodules haven't been initialized : ${UNINITIALISED_SUBMODS[@]}"
+    echo "Try running this first:"
+    echo "    git submodule update --init"
+    exit 2
+fi
+
+# Read the existing submodules information into an array. After this, the array
+# should contain an even number of entries, with each pair of entries containing
+# the submodule name and the current submodule version respectively.
+ARR=(`git submodule | sed 's|^.*submodules/\(.*\) (\(.*\))$|\1 \2|g'`)
```

Comment:
Ha, this is the line that makes it depends on being in `submodules/`? But this is not really needed to be part of the regex, right?